File: java/chopin.java
import java.util.Scanner;

public class chopin {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int cases = 1;
		while (scan.hasNext()) {
			String note = scan.next();
			String tonality = scan.next();
			if (note.length() == 1)
				System.out.println("Case " + cases++ + ": UNIQUE");
			else {
				int inc = note.charAt(1) == '#' ? 1 : -1;
				char ch = (char) ((note.charAt(0) + inc) % ('G' + 1));
				if (note.charAt(0) == 'G' && inc == 1)
					ch = 'A';
				else if (note.charAt(0) == 'A' && inc == -1)
					ch = 'G';
				System.out.printf("Case %d: %c%c %s%n", cases++, ch, (note.charAt(1) == '#' ? 'b' : '#'), tonality);
			}
		}
		scan.close();
	}
}

File: ts/src/chopin.ts
import readline = require('readline')

let lineCount = 0

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    lineCount++

    let [note, tonality] = line.split(' ')
    note ??= ''
    tonality ??= ''

    if (note.length === 1) {
      console.log(`Case ${lineCount}: UNIQUE`)
    } else {
      let ch: string
      switch (note) {
        case 'G#':
          ch = 'A'
          break
        case 'Ab':
          ch = 'G'
          break
        default:
          ch = String.fromCharCode(
            note.charCodeAt(0) + (note.charAt(1) === '#' ? 1 : -1)
          )
          break
      }
      console.log(
        `Case ${lineCount}: ${ch}${
          note.charAt(1) === '#' ? 'b' : '#'
        } ${tonality}`
      )
    }
  })
  .on('close', () => {
    process.exit(0)
  })
